solutions/swb-app/src/environmentTypeConfigRoutes.ts

```diff
@@ -47,6 +54,48 @@ export function setUpEnvTypeConfigRoutes(
     })
   );
 
+  // Soft Delete envTypeConfig
+  router.delete(
+    '/environmentTypes/:envTypeId/configurations/:envTypeConfigId',
+    wrapAsync(async (req: Request, res: Response) => {
+      const envTypeConfigDeleteRequest = {
+        envTypeId: req.params.envTypeId,
+        envTypeConfigId: req.params.envTypeConfigId
+      };
+      const validatedRequest = validateAndParse<DeleteEnvironmentTypeConfigRequest>(
+        DeleteEnvironmentTypeConfigRequestParser,
+        envTypeConfigDeleteRequest
+      );
+
+      async function checkDependency(envTypeId: string, envTypeConfigId: string): Promise<void> {
+        const typeId = `${resourceTypeToKey.envType}#${envTypeId}${resourceTypeToKey.envTypeConfig}#${envTypeConfigId}`;
+        const dependencies = await metadataService.listResourceByDependency<EnvironmentItem>(
```

Comment:
Do we need to list environments? I know it's more user friendly, unless @nguyen102 opposed to listing